src/lib/cli.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * @fileoverview Main CLI object, it reads the options from cli.
+ */
+
+/*
+ * The CLI object should *not* call process.exit() directly. It should only return
+ * exit codes. This allows other programs to use the CLI object and still control
+ * when the program exits.
+ */
+
+import * as path from 'path';
+
+import { debug as d } from './utils/debug';
+import * as globby from 'globby';
+import chalk from 'chalk';
+
+const debug: debug.IDebugger = d(__filename);
+
+import { options } from './cli/options';
+import { CLIOptions, IMDFile, ILink } from './types';
+import { MDFile } from './mdfile';
+
+const getMDFiles = async (directory): Promise<IMDFile[]> => {
+    const filesPath = await globby('**/*.md', { cwd: directory });
+
+    const result = [];
+
+    filesPath.forEach((relativePath) => {
+        const file = new MDFile(directory, relativePath);
+
+        result.push(file);
+    });
+
+    return result;
+};
+
+const validateLinks = async (mdFiles: IMDFile[]): Promise<void> => {
+    const promises: Promise<void>[] = [];
+
+    for (const mdFile of mdFiles) {
```

Comment:
@molant The interior `await` makes using `map` tricky if we want to fully process these one-at-a-time (which the comment on the next line suggests we need to do to avoid errors). I'm good with this as-is.